fmlcore/src/main/java/net/minecraftforge/fml/ModLoadingContext.java

```diff
@@ -46,10 +51,24 @@ public <T extends Record & IExtensionPoint<T>> void registerExtensionPoint(Class
     }
 
     public void registerConfig(ModConfig.Type type, IConfigSpec<?> spec) {
+        if (spec.isEmpty())
+        {
+            // This handles the case where a mod tries to register a config, without any options configured inside it.
+            LOGGER.debug("Attempted to register an empty config for type {} on mod {}", type, getActiveContainer().getModId());
+            return;
+        }
+
         getActiveContainer().addConfig(new ModConfig(type, spec, getActiveContainer()));
     }
 
     public void registerConfig(ModConfig.Type type, IConfigSpec<?> spec, String fileName) {
+        if (spec.isEmpty())
+        {
+            // This handles the case where a mod tries to register a config, without any options configured inside it.
+            LOGGER.debug("Attempted to register an empty config for type {} on mod {} using file name {}", type, getActiveContainer().getModId(), fileName);
+            return;
+        }
+
```

Comment:
These changes *can* and most probably will be considered breaking changes, so including it in an unrelated PR is not a good idea. We should try to avoid cases like #9061.

Reply:
what mod is creating an empty config? I cannot see any practical usecase.

Reply:
This was requested by lex.